### DISCUSSION TITLE
Is there an example with 3D geometry and how to set parameter function

### DISCUSSION THREAD

**Discussion by deng-cy** [pybamm-team/PyBaMM#2608] [Q&A]
Hi, I am interested in building an 3D electrode with porosity as a function of space. I do not know how to build a 3D model, so I am looking for examples. For parameter function, I saw a [tutorial](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/Getting%20Started/Tutorial%204%20-%20Setting%20parameter%20values.ipynb) which sets the current as a function of time; it seems to me that the input variable, time, is somehow preset. For other parameters, how should I know what input variables are used in the function? 

Any advice will be appreciated!

**Comment by valentinsulzer** [comment]
You can look at this file to see what inputs are required https://github.com/pybamm-team/PyBaMM/blob/dad2b10d56c1c56330722935569aca47393d552e/pybamm/parameters/lithium_ion_parameters.py#L464-L466 The second argument to `FunctionParameter` is a dictionary of the required inputs. You have to either give a function that takes all the inputs in the correct order, or just give a constant value. 

However, PyBaMM does not resolve the 3D microscale geometry. We use homogenized models instead as explained in [this paper](https://iopscience.iop.org/article/10.1088/2516-1083/ac7d31/pdf).

**Comment by deng-cy** [comment]
@tinosulzer Thanks for your reply. I am not looking for microscale geometry. I also need a homogenized model yet it is 3D. If 3D is not supported, is there a tutorial/example for 2D geometry?

**Reply by valentinsulzer** [reply]
See https://github.com/rtimms/asymptotic-pouch-cell

**Reply by deng-cy** [reply]
> See https://github.com/rtimms/asymptotic-pouch-cell

Thanks! I went through the example. My understanding is that it is a 1D+1D model, the second dimension z only works at current collector. The electrode does not have species flux along z direction. Is that correct? If so, is there a real 2D geometry model?

**Reply by valentinsulzer** [reply]
There is no real 2D geometry model, but our research shows that the 1+1D model is accurate, compared to a 2D COMSOL model, due to the small aspect ratio of the cell (tall and thin). See references in the linked repository
